### DISCUSSION TITLE
Suggestions on how to translate a model from paraview/pyvista to flopy/MODFLOW

### DISCUSSION THREAD

**Discussion by vcantarella** [modflowpy/flopy#2093] [Q&A]
Hi,

Although flopy has a method to export to VTK and plot results in pyvista, but I am wondering if anyone has experience with the other way around.
I am doing some processing on a 3D structured grid, with paraview, which I plan to make a modflow model from, and I was wondering if anyone has tried building a flopy model from a grid made in paraview/pyvista and could share that experience.

Thanks!!

Vitor,

**Comment by RyanConway91** [comment]
Commenting so this stays on my radar... no real advice though. 

Here is an idea for potential workflow to qaqc this process:

1. make a disu (octree) using gridgen.
2. make a disu model from the gridgen
```
gridprops = g.get_gridprops_disu6()
disu = fp.mf6.ModflowGwfdisu(model = gwf,**gridprops) 
```

4. save it to vtk
5. reload the vtk and see if you can rebuild the gridprops correctly to make a disu

**Comment by sebmeng** [comment]
I also comment to see what answers are posted. Would be very useful to have a way to go from vtk to a MODFLOW grid!

**Comment by wpbonelli** [comment]
I could see a case for classmethods on `flopy.discretization.grid.Grid` e.g. `Grid.from_vtk(mesh)`, `Grid.from_pyvista(mesh)`. Or maybe just one since [pyvista meshes subclass vtk](https://tutorial.pyvista.org/tutorial/06_vtk/a_2_pyvista_vtk.html#pyvista-and-vtk-together)?

Doing the translation seems nontrivial and we will have to reject some inputs since VTK and pyvista meshes can have nonlinear cells
